packages/cozy-client/src/CozyClient.spec.js

```diff
@@ -1381,6 +1381,75 @@ describe('CozyClient', () => {
       expect(client).toMatchSnapshot()
     })
   })
+
+  describe('fetchQueryAndGetFromState', () => {
+    const setup = () => {
+      const client = new CozyClient({})
+      const fileCol = new FileCollection('io.cozy.files', client.stackClient)
+      client.stackClient.collection.mockReturnValue(fileCol)
+      client.stackClient.fetchJSON = jest.fn().mockResolvedValue({
+        data: {
+          _id: 'resultFromQuery'
+        }
+      })
+      getQueryFromState.mockReturnValue({
+        data: {
+          _id: 'resultFromState'
+        }
+      })
+      return { client }
```

Comment:
Nit: Have you seen that could have a mock client where you can specify which queries are already in the state ? It has been implemented for test. The fetchJSON method is also mocked.

Reply:
oui j'ai vu mais justement il me semblait que ça posait problème pour avoir la réponse d'une query et tester fetchJSON... j'ai peut-être mal fait mon test quand j'ai essayé avec createMockClient. Je garde ça en tête pour le prochain.